awsgi/__init__.py

```diff
@@ -31,12 +32,21 @@ def response(self, output):
             'headers': dict(self.headers),
             'body': self.body.getvalue(),
         }
-        for chunk in output:
-            resp['body'] += chunk
+
+        if output:
+            body = bytearray(resp['body'])
+            [body.extend(chunk) for chunk in output]
+
+            resp['body'] = base64.b64encode(body)
+            resp['isBase64Encoded'] = True
+
         return resp
 
 
 def environ(event, context):
+    if event.get('isBase64Encoded', False) and event.has_key('body'):
+        event['body'] = base64.b64decode(event['body'])
```

Comment:
...and API Gateway will only encode if the content-type matches the list of binary types (`*/*` in my case, but that doesn't need to be true), otherwise it will be raw text.